src/object/srv_ec.c

```diff
@@ -0,0 +1,296 @@
+/**
+ * (C) Copyright 2016-2019 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ "
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+/**
+ * This file is part of daos_sr
+ *
+ * src/object/srv_ec.c
+ */
+#define D_LOGFAC	DD_FAC(object)
+
+#include <stddef.h>
+#include <stdio.h>
+#include <daos/rpc.h>
+#include <daos_types.h>
+#include "obj_rpc.h"
+#include "obj_internal.h"
+
+/* Determines if entire update affects just this target. If so, no IOD
+ * modifications or special bulk-transfer handling are needed for this
+ * update request.
+ */
+static bool
+ec_is_one_cell(daos_iod_t *iod, struct daos_oclass_attr *oca,
+	       unsigned int tgt_idx)
+{
+	unsigned int    len = oca->u.ec.e_len;
+	unsigned int    k = oca->u.ec.e_k;
+	bool		rc;
+	unsigned int	j;
+
+	for (j = 0; j < iod->iod_nr; j++) {
+		daos_recx_t     *this_recx = &iod->iod_recxs[j];
+		uint64_t         recx_start_offset = this_recx->rx_idx *
+						     iod->iod_size;
+		uint64_t         recx_end_offset =
+					(this_recx->rx_nr * iod->iod_size) +
+					recx_start_offset;
+
+		if (recx_start_offset & PARITY_INDICATOR) {
+			rc = false;
+			break;
+		} else if (recx_start_offset/len == recx_end_offset/len &&
+			(recx_start_offset % (len * k)) / len == tgt_idx) {
```

Comment:
I think I put it in like this because an IOD list can contain multiple recx entries per IOD.  Some of the recx entries may not touch this target.  The request is routed here if at least one recx maps to this target.